### ISSUE TITLE
whenReachable() and whenUnreachable() do not work

### ISSUE THREAD

**Issue by tiana890** [ashleymills/Reachability.swift#144]
Hello. Can you help me with my problem? I have always used this framework in my apps but now I can't figure out what I am doing wrong.

In my viewDidLoad() function I call this method:

    //MARK: Reachability
    func startReachability(){
    //declare this property where it won't go out of scope relative to your listener
    do{
        let reachability = try Reachability.reachabilityForInternetConnection()

        reachability.whenReachable = { reachability in
            // this is called on a background thread, but UI updates must
            // be on the main thread, like this:
            dispatch_async(dispatch_get_main_queue()) {
                if reachability.isReachableViaWiFi() {
                    print("Reachable via WiFi")
                } else {
                    print("Reachable via Cellular")
                }
            }
        }
        reachability.whenUnreachable = { reachability in
            // this is called on a background thread, but UI updates must
            // be on the main thread, like this:
            dispatch_async(dispatch_get_main_queue()) {
                print("Not reachable")
            }
        }

        try reachability.startNotifier()

    } catch {
        print("Unable to start notifier")
    }
    }

But it does not work. It calls whenReachable() and whenUnreachable() only once and when I turn Wi-Fi off and on it does nothing. I use 2.4 version.

**Comment by pawlusmall** [comment]
Same for me using iOS 10 and debugging on a real device.

**Comment by ashleymills** [comment]
You need a strong reference to your `reachability` variable, so declare it at class level. See the example in the readme, issues #93, #85

**Comment by tiana890** [comment]
@ashleymills thank u very much) it is evident but sometimes shit happens)))
